daemon/containerd/image_push.go

```diff
@@ -164,16 +171,48 @@ func (i *ImageService) pushRef(ctx context.Context, targetRef reference.Named, m
 
 	err = remotes.PushContent(ctx, pusher, target, store, limiter, platforms.All, handlerWrapper)
 	if err != nil {
-		if containerdimages.IsIndexType(target.MediaType) && cerrdefs.IsNotFound(err) {
-			return errdefs.NotFound(fmt.Errorf(
-				"missing content: %w\n"+
-					"Note: You're trying to push a manifest list/index which "+
-					"references multiple platform specific manifests, but not all of them are available locally "+
-					"or available to the remote repository.\n"+
-					"Make sure you have all the referenced content and try again.",
-				err))
```

Comment:
Extracted the message to the CLI by emitting extra Aux progress:
https://github.com/docker/cli/pull/4984/commits/dd83d4f4d82c89d634ccc2da9730753cb317406c#diff-f6f762242d9c434bd04b843b61984e2c9cf4a5b9a4086dff2390c3c7b7edbb22R162-R173